inst/HTM0/HTM0_whyUseMizer.Rmd

```diff
@@ -77,108 +79,112 @@ We can do the same with Biomass to see when, if any, of the species collapse. Fo
 
 ```{r}
-effort<-melt(sim@effort)  %>% filter(time >= 1980)
+effort <- melt(getEffort(sim))  %>% filter(time >= 1980)
 
-plot_effort<-ggplot(effort) + 
-geom_line(aes(x = time, y = value, colour = gear))+ 
-theme_pubr() +
-labs(y="Effort",x="") 
+plot_effort <- ggplot(effort) + 
+    geom_line(aes(x = time, y = value, colour = gear)) + 
+    theme_pubr() +
+    labs(y = "Effort", x = "") 
 
 plot_effort
 ```
 
 The species' maximum fishing mortality rates (which we call for convenience effort) for Cod and Saithe have declined but are not as low as what would be in line with single-species "Fmsy" (a reference level of fishing  deemed sustainable from  single-species stock assessements) and the fishing mortality rate for Sprat and Dab is very high. 
 
-We can the values directly like this:
+The effort values in 2018 were
 
 ```{r}
-current_effort<-sim@effort["2018",]
+(current_effort <- getEffort(sim)["2018", ])
 ```
 
 
 Let's start a new simulation that begins with the effort from 2018 and projects forward for 50 years.
 We will apply a linear reduction in effort for a selected species to a target value (here assumed for simplicity to be 0.2 with effort expressed in terms of the species' fishing mortality rate for fully selected sizes).
 
-To do this need to work with the effort array (time x gear) to enable changes in effort through time, for this scenario. Here we have a 'gear' for each species, since effort in this case were annual single-species fishing mortality estimates.
+To do this we need to work with the effort array (time x gear) to enable changes in effort through time, for this scenario. Here we have a 'gear' for each species, since effort in this case were annual single-species fishing mortality estimates.
 
 
 ```{r}
-sim_status_quo<-project(params,initial_n = sim@n["2018",,], initial_n_pp = sim@n_pp["2018",],effort=current_effort,t_max = 50)
+proj_effort <- matrix(current_effort, nrow = 51, ncol = 12, byrow = TRUE)
+dimnames(proj_effort) <- list(time = 2019:2069,
+                              gear = unique(gear_params(params)$gear))
 
-proj_effort<-sim_status_quo@effort
-
-target<-0.2
-select_species="Sprat"
+target <- 0.2
+sp_idx <- which(species_params(params)$species == "Sprat")
 
 # reach target by 10 years
-proj_effort[1:10,select_species]<-seq(from = proj_effort[1,select_species], to = target, length = 10)
+proj_effort[1:10, select_species] <- 
+    seq(from = proj_effort[1, select_species], to = target, length = 10)
 
 # then hold at target
-proj_effort[11:51,select_species]<-target
+proj_effort[11:51, select_species] <- target
 
 # check it
-proj_effort[,select_species]
+proj_effort[, select_species]
 
 # run the simulation forward, using the 2018 abundances as initial values
-
-sim_scen<-project(params,initial_n = sim@n["2018",,], initial_n_pp = sim@n_pp["2018",],effort=proj_effort,t_max = 50)
+params_status_quo <- setInitialValues(params, sim)
```

Comment:
Note that `setInitialValues()` is convenient for starting a new simulation from the final state of some other simulation.

Reply:
I like this and all of the other functions you have added Gustav - thank you!